### PR TITLE
Fix unwanted alignment after comment

### PR DESCRIPTION
**Pull request by Julow**
This was found when looking at the regressions of https://github.com/ocaml-ppx/ocamlformat/pull/2314

This removes the alignment in expressions like:

    let _ = try_with (fun () -> (* comment before *)
                                a ; b (* after b *) )

But this also adds regressions. This used to fit on one line:

    try%lwt
      (* fooooooooooo *)
      Lwt.return 2
    with _ -> assert false

I don't think it's worth supporting this again as the inline comment reduces readability and would break again with slight modifications of the code.

Comments after the `then` keyword now break:

    let foo =
      if cmp < 0 then
        (* ast higher precedence than context: no parens *)
        false
      else if cmp > 0 then
        (* context higher prec than ast: add parens *)
        true
      else ...

This could be supported again if needed. It's changed because it depended on implicit rules.

### CODE REVIEW

File: test/passing/tests/comments-no-wrap.ml.ref [hunk 1]
```diff
@@ -252,8 +252,11 @@ let%map
     _ =
   ()
 
-type t = < (* a *)
-         a: int [@atr] (* b *) ; b: int (* c *) >
+type t =
+  < (* a *)
+  a: int [@atr] (* b *)
+  ; b: int
+  (* c *) >
```
Comment (Julow) [inline test/passing/tests/comments-no-wrap.ml.ref:259]: This is equally broken as before. I wouldn't slow this PR down due to this.

File: test/passing/tests/comments-no-wrap.ml.ref [hunk 2]
```diff
@@ -296,8 +301,7 @@ type t =
 
 type t =
   (* | B *)
-  | A
-  (* A *)
+  | A (* A *)
```
Comment (Julow) [inline test/passing/tests/comments-no-wrap.ml.ref:304]: This is a fix, the comment is placed as in the input.

File: test/passing/tests/comments_args.ml.ref [hunk 3]
```diff
@@ -8,7 +8,8 @@ let emit_wrapper_function =
     false (* is_pair_generator *)
     hoisted true (* no_injection *)
     true (* inout_wrapper *)
-    is_interceptable false (* is_memoize_impl *)
+    is_interceptable false
+    (* is_memoize_impl *)
```
Comment (Julow) [inline test/passing/tests/comments_args.ml.ref:12]: This is a fix, the comment is placed as in the input.

File: test/passing/tests/ite-compact_closing.ml.ref [hunk 4]
```diff
@@ -152,3 +154,14 @@ let _ =
...
+let _ =
+  if
+    (* foo *)
+    foo
+  then 0
+  else if
+    (* bar *)
+    bar
+  then 1
+  else 2
```
Comment (Julow) [inline test/passing/tests/ite-compact_closing.ml.ref:167]: Before this PR, this formatted as (with every profiles):
```
let _ = if (* foo *)
           foo then 0 else if (* bar *)
                              bar then 1 else 2
```

File: test/passing/tests/js_source.ml.ocp [hunk 5]
```diff
@@ -113,7 +113,8 @@ let () =
   [%foo new x [@foo]];
   [%foo
     match[@foo] () with
-    | [%foo? (* Pattern expressions *)
+    | [%foo?
+             (* Pattern expressions *)
```
Comment (Julow) [inline test/passing/tests/js_source.ml.ocp:117]: Note: This is ocp-indent output.

File: test/passing/tests/let_module-sparse.ml.ref [hunk 6]
```diff
@@ -57,5 +57,6 @@ let () =
 
 let f () =
   let module (* comment *)
-  M = struct end in
+  M =
+  struct end in
```
Comment (Julow) [inline test/passing/tests/let_module-sparse.ml.ref:61]: This is a regression.

File: test/passing/tests/module.ml [hunk 7]
```diff
@@ -111,9 +111,11 @@ let _ =
   let module M =
     ( val (* aa *) m (* bb *)
         : (* cc *)
-        M with type t = t (* dd *)
+        M
+        with type t = t (* dd *)
```
Comment (Julow) [inline test/passing/tests/module.ml:115]: This is a regression.

File: test/passing/tests/let_module-sparse.ml.ref (hunk 6 above)
Comment (Julow) [reply test/passing/tests/let_module-sparse.ml.ref:75]: Fixed. Now respect the input more.

File: test/passing/tests/ite-kw_first_no_indicate.ml.ref [hunk 8]
```diff
@@ -159,3 +161,12 @@ let _ =
   else
     fun xxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyyyyyy zzzzzzzzzzz ->
     xxxxxxxxx yyyyyyyyyy zzzzzzzzzzzz
+
+let _ =
+  if (* foo *)
+     foo
+  then 0
+  else if (* bar *)
+          bar
```
Comment (gpetiot) [inline test/passing/tests/ite-kw_first_no_indicate.ml.ref:170]: This one could be improved.

File: test/passing/tests/ite-kw_first_closing.ml.ref [hunk 9]
```diff
@@ -175,3 +177,12 @@ let _ =
   else
     fun xxxxxxxxxxxxxxxxx yyyyyyyyyyyyyyyyyyyy zzzzzzzzzzz ->
     xxxxxxxxx yyyyyyyyyy zzzzzzzzzzzz
+
+let _ =
+  if (* foo *)
+     foo
+  then 0
+  else if (* bar *)
+          bar
```
Comment (gpetiot) [inline test/passing/tests/ite-kw_first_closing.ml.ref:186]: This one could be improved.

File: test/passing/tests/ite-kw_first_no_indicate.ml.ref (hunk 8 above)
Comment (Julow) [reply test/passing/tests/ite-kw_first_no_indicate.ml.ref:170]: This seems to be the intended behavior.

Comment (gpetiot) [reply test/passing/tests/ite-kw_first_no_indicate.ml.ref]: After having a look at the diff, I'm not sure it looks better on "real" code, we can discard this commit.

File: test/passing/tests/module.ml (hunk 7 above)
Comment (gpetiot) [reply test/passing/tests/module.ml]: Fixed in cea6493

**Review by gpetiot — APPROVED** [review]
LGTM, thanks!

08934b4 is optional though

**Comment by Julow** [comment]
Has 08934b4 changes code that do not contain comments, could you open a new PR for it ? That would improve debugging later.

**Comment by Julow** [comment]
Many thanks for your help! I was pulling my hairs out on this one.

**Pull request merged**